### PR TITLE
now.uiowa.edu Article Type site split

### PR DESCRIPTION
**Pull request by cory-skeers**
Relates to the first slice of #5474 

# How to test

`git checkout now-news && git pull`
`blt ds --site=now.uiowa.edu`
`drush @now.local en now_core smart_date`
`drush @now.local cim --source ../config/sites/now.uiowa.edu --partial`
`drush @now.local cim`
`drush @now.local cr`

Try adding some new articles with the various article types. Article Type should be required, defaulting to News Feature.

- `In the News` adds an "Original Publication Date" field
- `Achievement` currently matches the default Article type
- `News Feature` adds an "Embargo Message" field
  - If there is content in the Embargo Message field, the news feature should not be able to be published.

### CODE REVIEW

**Comment by cory-skeers** [comment]
Setting back to draft for a couple updates, accounting for using smart_date, hide embargo message field on display, possibly update original publication date display.

**Comment by bspeare** [comment]
@cory-skeers tried this out and got this message at the top of the node add screen: `Notice: Undefined index: #type in Drupal\Core\Form\FormHelper::processStates() (line 211 of core/lib/Drupal/Core/Form/FormHelper.php).`

**Comment by cory-skeers** [comment]
> @cory-skeers tried this out and got this message at the top of the node add screen: `Notice: Undefined index: #type in Drupal\Core\Form\FormHelper::processStates() (line 211 of core/lib/Drupal/Core/Form/FormHelper.php).`

Thanks for the heads up! I think some form config didn't get updated correctly, so then it was trying to apply `#states` to disabled fields. Should be fixed for now, but will need to check for that again after updating the original publication date field.

**Comment by cory-skeers** [comment]
> > @cory-skeers tried this out and got this message at the top of the node add screen: `Notice: Undefined index: #type in Drupal\Core\Form\FormHelper::processStates() (line 211 of core/lib/Drupal/Core/Form/FormHelper.php).`
> 
> Thanks for the heads up! I think some form config didn't get updated correctly, so then it was trying to apply `#states` to disabled fields. Should be fixed for now, but will need to check for that again after updating the original publication date field.

This is still an issue on import. The form display config looks correct on export on initial skim, but on import the new fields are disabled and need to be manually updated.

**Comment by joewhitsitt** [comment]
>This is still an issue on import. The form display config looks correct on export on initial skim, but on import the new fields are disabled and need to be manually updated.

I am not seeing config related to 'core.entity_form_display.node.article.*' in this PR. is this just on your local?

**Comment by cory-skeers** [comment]
> I am not seeing config related to 'core.entity_form_display.node.article.*' in this PR. is this just on your local?

`config_split.patch.core.entity_form_display.node.article.default.yml`? I'd set it up as a partial split, thinking it would only need to adjust weights and add the new fields.

**Comment by joewhitsitt** [comment]
~~right but the resulting config (patch or full) is not present in the PR~~

Update. sry. gosh. my bad.

https://github.com/uiowa/uiowa/pull/5501/files#diff-a18715ef3d4dacfebb5337ca35b0aaca8baa88a3e7c3f8b14eec6ab1f4902fc4

**Comment by joewhitsitt** [comment]
@cory-skeers i made some tweaks and updated the test instructions. had to run cim again after the partial import

**Comment by cory-skeers** [comment]
> @cory-skeers i made some tweaks and updated the test instructions. had to run cim again after the partial import

Thank you @joewhitsitt! Tried it out with your adjustments and worked like a charm.

**Comment by bspeare** [comment]
@cory-skeers is it best practice to have a `.install` file in the module?  I notice some sites have it and some don't.

File: config/default/core.entity_form_display.node.article.default.yml [hunk 1]
```diff
@@ -113,7 +114,9 @@ content:
     type: metatag_firehose
     weight: 6
     region: content
-    settings: {  }
+    settings:
```
Comment (bspeare) [inline config/default/core.entity_form_display.node.article.default.yml:117]: What does this new addition add?  Just curious.

**Comment by cory-skeers** [comment]
> @cory-skeers is it best practice to have a `.install` file in the module? I notice some sites have it and some don't.

That's a good question...I tend to include it as a personal practice, but not sure if there is a best practice. I might do some digging around d.o to see if there's a common community practice to follow.

**Review by bspeare — APPROVED** [review]
All the testing steps worked for me!

**Comment by briand44** [comment]
> > @cory-skeers is it best practice to have a `.install` file in the module? I notice some sites have it and some don't.
> 
> That's a good question...I tend to include it as a personal practice, but not sure if there is a best practice. I might do some digging around d.o to see if there's a common community practice to follow.

I doubt it's needed unless there is something happening during install but that is just my guess.

File: config/default/core.entity_form_display.node.article.default.yml (hunk 1 above)
Comment (joewhitsitt) [reply config/default/core.entity_form_display.node.article.default.yml:117]: this was just a re-export of default. 🤷

**Comment by joewhitsitt** [comment]
@cory-skeers i saw the smart date unset stuff. is that really needed? grad college has a thesis defense date inline widget with duration 0

Screen Shot 2022-08-03 at 3 10 08 PM

I guess we never solved https://github.com/uiowa/uiowa/issues/2691 before adding more implementations...

**Comment by cory-skeers** [comment]
> @cory-skeers i saw the smart date unset stuff. is that really needed? grad college has a thesis defense date inline widget with duration 0
> 
> Screen Shot 2022-08-03 at 3 10 08 PM
> 
> I guess we never solved #2691 before adding more implementations...

It's a little goofy, but the way I went about it for this case where only a date is needed, was to use the date only range widget and hide the end date (and the extra "to" that gets displayed with it). 

image

I don't think it's absolutely necessary, but with the current implementation of smart_date, it seems like a cleaner editor interface without preventing us from switching widgets later.

I'll add a `@todo` that references the issue so it doesn't get lost.

**Pull request merged**